master/buildbot/scripts/populatedatabase.py

```diff
@@ -0,0 +1,206 @@
+    )
+
+
+def load_config(config, config_file_name='master.cfg'):
+    if not config['quiet']:
+        print("checking %s" % config_file_name)
+
+    try:
+        master_cfg = config_module.MasterConfig.loadConfig(
+            config['baseDir'],
+            config_file_name,
+        )
+    except config_module.ConfigErrors as e:
+        print("Errors loading configuration:")
+        for msg in e.errors:
+            print("  " + msg)
+        return
+    except Exception:
+        print("Errors loading configuration:")
+        traceback.print_exc(file=sys.stdout)
+        return
+
+    return master_cfg
+
+
+@defer.inlineCallbacks
+def populate_user(db, user_count, verbose=True):
+    """
+    This function create `user_count` number of random user in database
+    :param db: a handler to the DBConnection object
+    :param user_count: an integer value with number of new users
+    """
+    if verbose:
+        print("Starting creating users")
+    if user_count > MAX_UNIQUE_USER_COUNT:
+        raise ValueError("Can not generate more than %d unique user" % MAX_UNIQUE_USER_COUNT)
+
+    users = []
+    created = 0
+    unique_identifier = set()
+    for ind in range(user_count):
+        # generate random identifier
+        identifier = names.get_first_name()
+        attempt = 0
+        while identifier in unique_identifier:
+            identifier = names.get_first_name()
+            attempt += 1
+            if attempt > user_count:
+                raise RuntimeError("Can not find unique name. Please choose small amount of records")
+
+        unique_identifier.add(identifier)
+        user = {
+            'identifier': identifier,
+            'bb_username': identifier,
+            'bb_password': 'pyflakes'
+        }
+        result = yield db.users.createUser(user)
+        if result:
+            created += 1
+        users.append(user)
+        if verbose:
+            print_progress_bar(ind+1, user_count)
+
+    if verbose:
+        print("Created %d new users, %d skipped" % (created, user_count - created))
+    defer.returnValue(map(lambda x: x['identifier'], users))
+
+
+@defer.inlineCallbacks
+def populate_build(db, build_count, builders_list, projects, user_names, verbose=True):
+    """
+
+    :param db: a handler to the DBConnection object
+    :param build_count: an integer value with number of new builds
+    :param builders_list: a list of builders. The builder is a BuilderConfig object
+    :param projects: a list of a ProjectConfig objects
+    :param user_names: a list of an usernames (identifier) from the database
+    """
+    if verbose:
+        print("Starting creating builds")
+    created = 0
+    completed_results = [
+        results.SUCCESS,
+        results.WARNINGS,
+        results.FAILURE,
+        results.SKIPPED,
+        results.EXCEPTION,
+        results.CANCELED,
+        results.NOT_REBUILT,
+        results.DEPENDENCY_FAILURE,
+        results.MERGED,
+        results.INTERRUPTED,
+    ]
```

Comment:
Don't duplicate this list, perhaps you could create an ALL_RESULTS in `results.py`.

Reply:
Solved 👍